### PR TITLE
🐛 Fix OpenTelemetry endpoint URL creation

### PR DESCRIPTION
**Pull request by giancarloromeo**
## What do these changes do?
The migration to Pydantic v2 caused issues when building the endpoint to the OpenTelemetry service.

```
raise ConnectionError(e, request=request)\nrequests.exceptions.ConnectionError: HTTPConnectionPool(host='otel-collector', port=80): Max retries exceeded with url: /:4318/v1/traces (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ea65c17d510>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

As you can see from logs, it tried to connect to `/:4318/v1/traces`, which contains a / derived from new Pydantic v2 `AnyUrl` type.

YARL will be used for URL composition.

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]

File: packages/service-library/src/servicelib/aiohttp/tracing.py [hunk 1]
```diff
@@ -72,9 +73,9 @@ def setup_tracing(
     resource = Resource(attributes={"service.name": service_name})
     trace.set_tracer_provider(TracerProvider(resource=resource))
     tracer_provider: trace.TracerProvider = trace.get_tracer_provider()
-    tracing_destination: str = (
-        f"{opentelemetry_collector_endpoint}:{opentelemetry_collector_port}/v1/traces"
-    )
+
+    opentelemetry_collector_path = "/v1/traces"
```
Comment (GitHK) [inline packages/service-library/src/servicelib/aiohttp/tracing.py:77]: you can drop this var, not that useful

File: packages/service-library/src/servicelib/fastapi/tracing.py [hunk 2]
```diff
@@ -75,7 +76,14 @@ def setup_tracing(
     trace.set_tracer_provider(TracerProvider(resource=resource))
     global_tracer_provider = trace.get_tracer_provider()
     assert isinstance(global_tracer_provider, TracerProvider)  # nosec
-    tracing_destination: str = f"{tracing_settings.TRACING_OPENTELEMETRY_COLLECTOR_ENDPOINT}:{tracing_settings.TRACING_OPENTELEMETRY_COLLECTOR_PORT}/v1/traces"
+
+    opentelemetry_collector_endpoint: str = (
+        f"{tracing_settings.TRACING_OPENTELEMETRY_COLLECTOR_ENDPOINT}"
+    )
```
Comment (GitHK) [inline packages/service-library/src/servicelib/fastapi/tracing.py:82]: as well as this one

**Review by GitHK — APPROVED** [review]
Thanks, some minor notes

**Review by matusdrobuliak66 — APPROVED** [review]

File: packages/service-library/src/servicelib/aiohttp/tracing.py [hunk 3]
```diff
@@ -72,9 +73,8 @@ def setup_tracing(
     resource = Resource(attributes={"service.name": service_name})
     trace.set_tracer_provider(TracerProvider(resource=resource))
     tracer_provider: trace.TracerProvider = trace.get_tracer_provider()
-    tracing_destination: str = (
-        f"{opentelemetry_collector_endpoint}:{opentelemetry_collector_port}/v1/traces"
-    )
+
+    tracing_destination: str = f"{URL(opentelemetry_collector_endpoint).with_port(opentelemetry_collector_port).with_path('/v1/traces')}"
```
Comment (sanderegg) [inline packages/service-library/src/servicelib/aiohttp/tracing.py:77]: are you sure you need to put / in front of v1? well I guess it handles it.

**Review by sanderegg — APPROVED** [review]
thanks

File: packages/service-library/src/servicelib/aiohttp/tracing.py (hunk 3 above)
Comment (giancarloromeo) [reply packages/service-library/src/servicelib/aiohttp/tracing.py:77]: Yes, because there is a port. The problem was before.

**Pull request merged**